src/aatoolbox/utils/raster.py

```diff
@@ -0,0 +1,447 @@
+"""
+Utilities to manipulate and analyze raster data.
+
+This module extends xarray and rioxarray to provide
+additional functionality for raster processing and
+post-processing. The extension is based on the
+guidance for how to extend xarray:
+
+http://xarray.pydata.org/en/stable/internals/extending-xarray.html
+
+However, since rioxarray already extends xarray, this
+modules extensions inherit from the RasterArray and
+RasterDataset extensions respectively. This ensures
+cleaner code in the module as ``rio`` methods are
+available immediately, but also means a couple of
+design decisions are followed.
+
+The xarray.DataArray and xarray.DataSet
+extensions here inherit from rioxarray base classes.
+Thus, methods that are identical for both objects
+are defined in a mixin class ``AatRasterMixin`` which
+can be inherited by the two respective extensions.
+"""
+
+import logging
+from typing import Any, Callable, Dict, List, Union
+
+import geopandas as gpd
+import numpy as np
+import pandas
+import pandas as pd
+import rioxarray
+import xarray
+import xarray as xr
```

Comment:
You should only import xarray once

Reply:
So, interestingly, I think thiswas being pushed back in there during pre-commit, and I hadn't noticed. This is because I had set the return typing to be `xarray.DataArray` rather than `xr.DataArray`, which helped automatically generate the docstring without the `xr` alias.